tests/test_series.py

```diff
@@ -496,9 +525,40 @@ def test_types_compare() -> None:
 
 def test_types_agg() -> None:
     s = pd.Series([1, 2, 3], index=["col1", "col2", "col3"])
-    s.agg("min")
-    s.agg(x=max, y="min", z=np.mean)
-    s.agg("mean", axis=0)
+    check(s.agg("min"), np.int64)
+    check(s.agg(min), np.int64)
```

Comment:
add `assert_type` on the above 2

Reply:
I tried but had a small problem as the return type is `S1` and I couldn't get both mypy and pyright to pass (mypy wanted `Any` while pyright wanted `Unknown`).

Reply:
Maybe the return type in this case should just be `Scalar`?

Reply:
If you use `Scalar`, then it is a `Union` of types, so that isn't friendly.  `S1` is correct.  But when you create a `Series` as you did, it is generic, so tracked as `Series[Any]`.  I think you could do
```python
check(assert_type(s.agg("min"), Any), np.int64)
```
or maybe this:
```python
check(assert_type(s.agg("min"), "S1"), np.int64)
```

Reply:
or could cast `s` to be of type `Series[int]`?

Reply:
Just thought that `S1` is not entirely correct, for example if you aggregate using `"mean"` you can get floats from ints.
`Any` doesn't pass the pyright checks (expected "Any" but received "Unknown")
"S1" also doesn't seem to work (it's not defined).
Making `s` into a `Series[int]` creates problems because then all the type checks need to be compared against `pd.Series[int]` but then the pytest run fails because the "real" pandas Series type isn't generic.

Reply:
I think you could get `"S1"` to work by doing 
```python
if TYPE_CHECKING:
    from pandas._typing import S1
```

I agree that the result won't always be right, e.g., with "mean", but you could then create an overload that says if "mean" or `mean` are passed, the result is float.

Reply:
It doesn't work because `S1` is an unbound `TypeVar`.

Reply:
This is coming down to different behaviors with `mypy` and `pyright`.  I prefer the `mypy` behavior here.  I've raised an issue with `pyright` to see what they say.    https://github.com/microsoft/pyright/issues/3794

Reply:
It appears that this is a `pyright` bug that will get fixed in the "next release", which will probably happen within a week, so let's wait for that, and then you can do `assert_type()` versus `Any` with both `mypy` and `pyright`

Reply:
It's now fixed in pyright 1.1.266 so you should be able to do
```python
check(assert_type(s.agg("min"), Any), np.int64)
```
You should update `pyproject.yaml` to refer to that version.